### ISSUE TITLE
Difficulty reproducing the reported high-resolution results

### ISSUE THREAD

**Issue by LIU-Yuxin** [PeterYYZhang/UltraImageGen#3]
Thank you for releasing this interesting work.

I am trying to reproduce the image quality reported in the paper using the current main branch, the provided LoRA weights, and an attention mask generated from the build_mask notebook. So far, my results are substantially below the quality shown in the paper.

These are images generated using Flux 4 step model + 4 step lora

_“A futuristic AI shrine built in a canyon, worshippers offering digital prayers to holographic deities”_
LR:
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/eb39d49a-7a28-4e0c-a924-c6eb54b32a58" />
HR zoom-in:
<img width="1081" height="689" alt="Image" src="https://github.com/user-attachments/assets/bf935879-e862-489a-9c6f-6877350207f4" />

<img width="1072" height="681" alt="Image" src="https://github.com/user-attachments/assets/5d012f45-3de0-4868-b37e-01f5ac12983a" />

_“Serene mountain villagenestled in the Swiss Alps,traditional woodenchalets with flowerboxes, cobblestone pathswinding between houses,snow-capped peaks in thebackground, golden hourlighting, smoke risingfrom chimneys, villagersin traditional clothingwalking the streets, cozywarm atmosphere,detailed architecture,rustic charm.”_
LR:
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/849a6dcb-816f-4d87-ab8c-a3769d67497a" />
HR zoom-in:
<img width="1083" height="692" alt="Image" src="https://github.com/user-attachments/assets/47421ae4-1243-4dc1-9899-3eb6ea8b905c" />

<img width="1076" height="691" alt="Image" src="https://github.com/user-attachments/assets/370eea58-84cf-4004-a684-6ba458df5744" />

_“An elven queen wearing transparent silk in a fantasy character portrait”_
LR:
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/a3f89d4b-b700-4e19-a9cc-f6771eb29506" />
HR zoom-in:
<img width="261" height="301" alt="Image" src="https://github.com/user-attachments/assets/27151ce3-d886-43da-9490-5c623a5e1dea" />

<img width="395" height="369" alt="Image" src="https://github.com/user-attachments/assets/03fd6c8b-1a79-4857-8b44-192efc83421f" />

The HR output shows less aliasing, but I do not see a clear increase in fine detail over the LR image. The distorted human structures remain distorted, and the HR result also appears flatter and lower in contrast.
The base model + base lora test also show no better result.

I also accidentally used 4 step model + base lora, and the results do seems to be closer to the ones shown in the paper, I am not sure if you have wrongly named the release checkpoints.

<img width="861" height="799" alt="Image" src="https://github.com/user-attachments/assets/c0ee2c6e-a619-47cc-8d5c-51167ad3b692" />

<img width="758" height="737" alt="Image" src="https://github.com/user-attachments/assets/0494452c-beaf-4874-aef6-69f6804b2d0c" />

<img width="1045" height="943" alt="Image" src="https://github.com/user-attachments/assets/55d51bb0-dbd6-4a2b-a3a2-a14fc23e449a" />

On the other hand, using base model with 4 step lora would not work.
Here are the config and inference script I am using.
[run_inference_4step_base_lora.sh](https://github.com/user-attachments/files/31690184/run_inference_4step_base_lora.sh)
[inference_flux2.yaml](https://github.com/user-attachments/files/31690185/inference_flux2.yaml)

For reference, I am running on an SM90 GPU and have adjusted the Q/K block sizes to 64 and 128, respectively. Since the required attention mask is not included, my setup is not a strictly out-of-the-box run, and the generated mask may differ from the one used for the reported results.

Could you provide:

- One known-good end-to-end inference command, including the required mask and installation/runtime settings

- The prompt, seed, LR/HR sampling steps, guidance/CFG, and other relevant settings for a few representative visual results in the paper

- Confirmation of whether the HR stage is expected mainly to refine local texture while inheriting structural errors from the LR guide

This would help determine whether the discrepancy comes from my inference setup or reflects the expected limitations of the released model.

**Comment by PeterYYZhang** [comment]
Hi in the paper FLUX2's result is also not as good as flux1, I guess it is because that the distilled model are not easy to fine tune. You can give it a try using the FLUX1.dev implementation, which can be found in another branch (but make sure you are using the correct environment since that one is implemented using an older diffusers library. I retried myself that using FLUX1.dev's branch will show good results up to 4kx4k, we apologize that we lost some configurations since this work was first done in August 2025, and some important details are lost in a disk failure issue on our server, the checkpoints are all retrained.

For the FLUX2's problem, we recently found that after using around 5000 4K images to finetune the output projection layers of the model the results will be a lot better. 

The QK block sizes will influence the speed and I don't think it will influence the results.

base model with 4 step lora won't work. You need to train it again.

**Comment by LIU-Yuxin** [comment]
Thank you for your reply! I will try Flux1 version instead.
